### PR TITLE
[feature](BE)pad missed version with empty rowset

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes
If all replicas of one tablet are broken, user can use this http api to pad the missed version with empty rowset.
Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.65 seconds
 load time: 442 seconds
 storage size: 17123356277 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221213061754_clickbench_pr_62423.html

File: be/src/http/action/pad_rowset_action.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    } else {
+        HttpChannel::send_reply(req, HttpStatus::INTERNAL_SERVER_ERROR, result);
+    }
+}
+
+Status PadRowsetAction::_handle(HttpRequest* req) {
+    // Get tablet id
+    const std::string& tablet_id_str = req->param(TABLET_ID);
+    if (tablet_id_str.empty()) {
+        std::string error_msg = std::string("parameter " + TABLET_ID + " not specified in url.");
```
Comment (platoneko) [inline be/src/http/action/pad_rowset_action.cpp:49]: consider `fmt::format`

File: be/src/http/action/pad_rowset_action.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    // Get schema hash
+    const std::string& version_str = req->param(VERSION);
+    if (version_str.empty()) {
+        std::string error_msg = std::string("parameter " + VERSION + " not specified in url.");
+        return Status::InternalError(error_msg);
+    }
+
+    // valid str format
+    int64_t tablet_id = std::atol(tablet_id_str.c_str());
+    int32_t version = std::atoi(version_str.c_str());
```
Comment (platoneko) [inline be/src/http/action/pad_rowset_action.cpp:62]: start_version, end_version may not be equal

**Review by morningman — COMMENTED** [review]
missing sidebar

File: be/src/http/action/pad_rowset_action.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    std::unique_ptr<RowsetWriter> writer;
+    RETURN_IF_ERROR(tablet->create_rowset_writer(version, VISIBLE, NONOVERLAPPING,
+                                                 tablet->tablet_schema(), -1, -1, &writer));
+    auto rowset = writer->build();
+    rowset->make_visible(version);
+
+    std::vector<RowsetSharedPtr> to_add {rowset};
+    std::vector<RowsetSharedPtr> to_delete;
+    tablet->modify_rowsets(to_add, to_delete);
```
Comment (platoneko) [inline be/src/http/action/pad_rowset_action.cpp:95]: Must hold header_lock during `modify_rowsets`

File: be/src/http/action/pad_rowset_action.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    LOG(INFO) << "handle request result:" << result;
+    if (status.ok()) {
+        HttpChannel::send_reply(req, HttpStatus::OK, result);
+    } else {
+        HttpChannel::send_reply(req, HttpStatus::INTERNAL_SERVER_ERROR, result);
+    }
+}
+
+Status PadRowsetAction::check_param(HttpRequest* req) {
+    RETURN_IF_ERROR(check_one_param(req->param(TABLET_ID), TABLET_ID));
```
Comment (dataroaring) [inline be/src/http/action/pad_rowset_action.cpp:55]: Arguments is put in wrong position.

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by morningman — COMMENTED** [review]
I think we'd better use a `SQL` command to repair the tablet.
Because there may be lots of broken tablets.
And there is no way for user to get all broken tablets' id.
By using a `SQL` command like:

`repair table xxx partition(xx)`.

We can get all broken tablets internal and help user to repair all of them

**Comment by ByteYue** [comment]
> We can get all broken tablets internal and help user to repair all of them
Would it be too heavy for one pr? I think maybe we can mark it as one future work? We can afterwards provide one sql command which collects all broken tablets's information and then calls the pad http inside this pr in one transaction for consistency.

**Comment by morningman** [comment]
> > We can get all broken tablets internal and help user to repair all of them
> 
> Would it be too heavy for one pr? I think maybe we can mark it as one future work? We can afterwards provide one sql command which collects all broken tablets's information and then calls the pad http inside this pr in one transaction for consistency.

ok

**Pull request merged**